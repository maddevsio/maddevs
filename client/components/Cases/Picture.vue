<template>
  <picture>
    <source
      class="image"
      :class="{'box-shadow': boxShadow}"
      :srcset="[require(`../../assets/img/Cases/${pictureFolder}/webp/${fileName}.webp`) + ' ', require(`../../assets/img/Cases/${pictureFolder}/webp/${fileName}@2x.webp`) + ' 2x']"
      type='image/webp'>
    <img
      class="image"
      :class="{'box-shadow': boxShadow}"
      :src="[require(`../../assets/img/Cases/${pictureFolder}/${fileExtension}/${fileName}.${fileExtension}`)]"
      :srcset="[require(`../../assets/img/Cases/${pictureFolder}/${fileExtension}/${fileName}@2x.${fileExtension}`) + ' 2x']"
      :alt="alt"
      :loading="lazyLoading ? 'lazy' : ''"
    >
  </picture>
</template>


<script>
export default {
  name: 'Picture',
  props: {
    pictureFolder: {
      types: String,
      default: ''
    },
    fileName: {
      type: String,
      default: ''
    },
    fileExtension: {
      type: String,
      default: ''
    },
    alt: {
      type: String,
      default: ''
    },
    lazyLoading: {
      type: Boolean,
      default: false
    },
    boxShadow: {
      type: Boolean,
      default: false
    }
  }
};
</script>
<style lang="scss" scoped>
  .image {
    width: 100%;
    height: auto;
    display: block;
  }

  .box-shadow {
    box-shadow: 0px 2px 7px rgba(0, 0, 0, 0.05), 0px 5.47001px 41.35px rgba(0, 0, 0, 0.1);
  }
</style>
