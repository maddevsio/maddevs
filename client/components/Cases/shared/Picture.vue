<template>
  <picture>
    <source
      :class="{ 'box-shadow': shadow, 'border-radius': radius, 'grey-background': background, 'media_lazy': lazy }"
      v-bind="attrsSource"
      class="image"
      type="image/webp"
    >
    <img
      :class="{ 'box-shadow': shadow, 'border-radius': radius, 'grey-background': background, 'media_lazy': lazy }"
      v-bind="attrsImg"
      :alt="alt"
      :width="width"
      :height="height"
      data-testid="test-picture-img"
      class="image"
      @load="onImageLoad($event)"
    >
  </picture>
</template>

<script>
export default {
  name: 'Picture',
  props: {
    folder: {
      type: String,
      default: '',
    },

    file: {
      type: String,
      default: '',
    },

    extension: {
      type: String,
      default: '',
    },

    alt: {
      type: String,
      default: '',
    },

    lazy: {
      type: Boolean,
      default: true,
    },

    shadow: {
      type: Boolean,
      default: false,
    },

    radius: {
      type: Boolean,
      default: false,
    },

    width: {
      type: Number,
      default: 0,
    },

    height: {
      type: Number,
      default: 0,
    },

    background: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    attrsImg() {
      const path = `/images/Cases/${this.folder}/${this.extension}/${this.file}.${this.extension}`
      if (this.lazy) {
        return {
          'data-src': [this.$getMediaFromS3(path)],
          'data-srcset': [this.$getMediaFromS3(path)],
        }
      }
      return {
        src: [this.$getMediaFromS3(path)],
        srcset: [this.$getMediaFromS3(path)],
      }
    },

    attrsSource() {
      const path = `/images/Cases/${this.folder}/webp/${this.file}.webp `
      const path2x = `/images/Cases/${this.folder}/webp/${this.file}@2x.webp 2x`
      if (this.lazy) {
        return {
          'data-srcset': [this.$getMediaFromS3(path), this.$getMediaFromS3(path2x)],
        }
      }
      return {
        srcset: [this.$getMediaFromS3(path), this.$getMediaFromS3(path2x)],
      }
    },
  },

  methods: {
    onImageLoad(event) {
      if (event.target.classList.contains('grey-background')) {
        event.target.classList.remove('grey-background')
      }
    },
  },
}
</script>
<style lang="scss" scoped>
.image {
  width: 100%;
  height: auto;
  display: block;
}

.grey-background {
  background-color: $bgcolor--silver;
}

.box-shadow {
  box-shadow: 0px 2px 7px rgba(0, 0, 0, 0.05), 0px 5.47001px 41.35px rgba(0, 0, 0, 0.1);
}

.border-radius {
  border-radius: 3px;
}
</style>
