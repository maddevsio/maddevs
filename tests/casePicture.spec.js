import {
  mount
} from '@vue/test-utils';

import Picture from '@/components/Cases/Picture';

describe('Picture', () => {
  let wrapper;
  let props = {
    pictureFolder: 'nambafood',
    fileName: 'apple-iphone-xr-silver',
    fileExtension: 'png',
    alt: 'alt',
    lazyLoading: true,
    boxShadow: true
  };

  beforeEach(() => {
    wrapper = mount(Picture, {
      propsData: props
    });
  });

  // ------ IMPORTANT ----- //
  test('is Vue\'s instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('should pass correct props', () => {
    expect(wrapper.props()).toEqual(props);
  });
});
