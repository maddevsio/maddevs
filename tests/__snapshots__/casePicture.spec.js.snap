// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture renders correctly 1`] = `
<picture>
  <source
    class="image box-shadow"
    srcset="[object Object] ,[object Object] 2x"
    type="image/webp"
  />
   
  <img
    alt="alt"
    class="image box-shadow"
    loading="lazy"
    src="[object Object]"
    srcset="[object Object] 2x"
  />
</picture>
`;
